### PR TITLE
feature: beacon sidecar iterator

### PR DESCRIPTION
**Pull request by JackG-eth**
Iterator for new beacon EIP4844 type

Motivation
This will alllow us to grab n blobs for a given tx from the CL

it will also unblock https://github.com/paradigmxyz/reth/pull/7443

### CODE REVIEW

File: crates/rpc-types-beacon/src/sidecar.rs [hunk 1]
```diff
@@ -12,6 +13,41 @@ pub struct BeaconBlobBundle {
...
+    type IntoIter = IntoIter<BlobData>;
+
+    fn into_iter(self) -> Self::IntoIter {
+        self.data.into_iter()
+    }
+}
+
+/// Intermediate type for BlobTransactionSidecar matching
+pub struct SidecarIterator {
+    pub iter: IntoIter<BlobData>,
```
Comment (mattsse) [inline crates/rpc-types-beacon/src/sidecar.rs:28]:
```suggestion
    iter: IntoIter<BlobData>,
```

File: crates/rpc-types-beacon/src/sidecar.rs [hunk 2]
```diff
@@ -12,6 +13,41 @@ pub struct BeaconBlobBundle {
...
+    type Item = BlobData;
+    type IntoIter = IntoIter<BlobData>;
+
+    fn into_iter(self) -> Self::IntoIter {
+        self.data.into_iter()
+    }
+}
+
+/// Intermediate type for BlobTransactionSidecar matching
+pub struct SidecarIterator {
```
Comment (mattsse) [inline crates/rpc-types-beacon/src/sidecar.rs:27]: derive debug + clone

Comment (mattsse) [inline crates/rpc-types-beacon/src/sidecar.rs:27]: let's also implement iterator for this type

**Review by mattsse — CHANGES_REQUESTED** [review]
few nits

**Review by yash-atreya — APPROVED** [review]
lgtm!

**Review by Evalir — APPROVED** [review]

**Pull request merged**